cluster-operator/src/main/java/io/strimzi/operator/cluster/model/Utils.java

```diff
@@ -0,0 +1,127 @@
+                return defaultValue;
+            }
+            return Integer.parseInt(data.get(key));
+        } catch (NumberFormatException e)  {
+            String msg = " is corrupted";
+            if (data.get(key).isEmpty()) {
+                msg = " is empty";
+            }
+            throw new InvalidConfigMapException(key, msg);
+        }
+    }
+
+    public static boolean getBoolean(Map<String, String> data, String key, boolean defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map
+            return defaultValue;
+        } // java parses "truue" as false. We need to be more strict
+        if (data.get(key).equals("true")) {
+            return true;
+        } else if (data.get(key).equals("false")) {
+            return false;
+        } else {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static Storage getStorage(Map<String, String> data, String key) {
+        try {
+            if (data.get(key) == null) { // value is not set in config map
+                return new Storage(Storage.StorageType.PERSISTENT_CLAIM);
+            }
+            return Storage.fromJson(new JsonObject(data.get(key)));
+        } catch (Exception e) {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static String getNonemptyString(Map<String, String> data, String key, String defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map -> will be used default value
+            return defaultValue;
+        }
+        if (data.get(key).length() == 0) {
+            throw new InvalidConfigMapException(key, " is empty");
+        }
+        return data.get(key);
+    }
+
+    public static String getJsonCorruptionBlame(DecodeException de, String config) {
+        String token = "";
+        token = de.getMessage();
+        if (token.contains("Illegal unquoted character")) {
+            return "JSON quotation";
+        }
+        token = token.replace("Failed to decode: Unrecognized token '", "");
+        token = token.substring(0, token.indexOf("'"));
+
+        String[] entries = config.replace("{", "").replace("}", "")
+               .replace("\"", "").replace(" ", "")
+               .replace("\n", "").split(",");
+        int i;
+        String blame = "";
+        for (i = 0; i < entries.length; ++i) {
+            if (entries[i].split(":")[1].equals(token)) {
+                blame = entries[i].split(":")[0];
+                break;
+            }
+        }
+        return blame;
+    }
+
+    public static JsonObject getConfig(Map<String, String> data, String key) {
+        String config = data.get(key);
+        if (config == null) {
+            return null;
+        }
+        try {
+            Map<String, Object> map = new JsonObject(config).getMap();
+            Map<String, String> numMap = new HashMap<String, String>();
```

Comment:
I do not understand the purpose of the `numMap`, `boolMap` and `stringMap` and the secondary validation using the `getInteger` etc.methods. The `JsonObject` already validated these. If the `JsonObject` class is able to convert the value to `Integer` or `Boolean` it has to be `Integer` or `Boolean`. 

Why do we have to double check it afterwards? Can't we just remove the lines 109-120?

Reply:
At first, i thouhgt there are just number values in whatever-config, so it was designed to iterate all values and check if its number. Yes, we can delete this lines, but issue is, if I put value "string123" to some number key, JsonObject parses it ok.

Reply:
You mean if someone does something like this?
```json
{
"this-shoutld-be-a-number": "string123"
}
```

But in most cases you anyway don't know whether the key should be a number or string. So you cannot validate this on this level. (we actually never validate this for the configuration or metrics fields).